### ISSUE TITLE
[Bug] Block user functionality bugs

### ISSUE THREAD

**Issue by ankar84** [RocketChat/Rocket.Chat#18279]
### Description:

There is a few issues with **Block user** functionality.
That issue partly related to #16100

First issue - User can block himself. I don't know for what case that can be useful. 
Second and most important - User without any privileges in private chat can Block any user in that chat. And the most interesting thing about that is that not only blocked user will not be able to send a message to chat, but user who block too will not be able to send a message to chat.
Third issue - block user technic is not informative to blocked user and to administrator. Blocked user see only **Error** message from Rocket.Cat bot. Hi do not see, that hi was blocked and whom (to resolve that issue). And even administrator in servers logs do not see who and by who has blocked - only `server.js:204 System ➔ error Error sending message: Error: room_is_blocked` message appears in logs.

### Steps to reproduce to second issue:

1. User1 creates private chat
2. User1 adds User2 to that private chat
3. User2 (without owner or moderator permissions) Block User1
4. User1 and User2 both gets Error message when they try to send message to chat

### Expected behavior:

Block user function should be available only to privileged users in chat and not for any regular user.
Block user should not block messages for user who block another user.

### Actual behavior:

That screencast shows private chat with 2 users - **Карлан Антон Андреевич** - owner of chat and **Карлан Антон Андреевич (Test1)** - regular user.
**Карлан Антон Андреевич (Test1)** blocks **Карлан Антон Андреевич** and after that can't send messages to chat.
**Карлан Антон Андреевич** can't send messages to chat too.
![block_user](https://user-images.githubusercontent.com/4023037/87630529-f5dc2800-c73d-11ea-9698-7b959b026f60.gif)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.1.1
- Operating System: CentOS7
- Deployment Method: docker
- Number of Running Instances: 25
- DB Replicaset Oplog: Enabled
- NodeJS Version: v12.16.1
- MongoDB Version: 4.0.17

### Client Setup Information

- Desktop App or Browser Version: Desktop 2.17.10 and Mobile 4.8.1
- Operating System: Windows 10

### Relevant logs:

````

at packages/ddp-server/livedata_server.js:559:43
at Session.method (packages/ddp-server/livedata_server.js:689:23)
at packages/ddp-server/livedata_server.js:717:46
at packages/ddp-server/livedata_server.js:715:46
at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
at new Promise (<anonymous>)
at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1771:12)
at packages/ddp-server/livedata_server.js:719:19
at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
at executeSendMessage (app/lib/server/methods/sendMessage.js:71:16)
at MethodInvocation.sendMessage (app/lib/server/methods/sendMessage.js:105:10)
at canSendMessage (app/authorization/server/functions/canSendMessage.js:38:84)
=> awaited here:
at Function.Promise.await (/app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)
at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
at app/authorization/server/functions/canSendMessage.js:21:9
server.js:204 System ➔ error Error sending message: Error: room_is_blocked

````

**Comment by bbrauns** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/17400

**Comment by ankar84** [comment]
> Same: #17400

Thanks a lot, Bjorn!
~It's really strange, because I did a search on **block** word.~
Oh, I get it! I did a search on opened issues.
I did check on 4.3.2 and there is no Block user functionality in private chats, so I'm closing that issue as solved by #17406
